documentation/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.md

```diff
@@ -0,0 +1,263 @@
+
+- Users writing components will have a "dark mode compatible by default"
+  experience. By default, they will not need to think much about dark mode.
+- Color names in the codebase will be replaced with semantic names.
+- Frontend developers will have easy tools to visually test components in light
+  and dark mode.
+
+## Step-by-step implementation plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+The following plan requires approved designs and semantic color names. Each task
+is a discrete issue and pull request. The top-level "Work Streams" can be
+completed in parallel.
+
+1. **Work Stream A**: Implement the new color palette.
+
+   1. Create a `FORCE_DARK_MODE` feature flag which is "off" by default and
+      "switchable" in our staging environment. Add a `dark-mode` class to the
+      root HTML tag when this flag is enabled, and a `light-mode` class which is
+      set by default. This will not result in any visual changes.
+   2. The following steps can take place in parallel:
+
+      1. Rename all colors in the Tailwind configuration and the frontend
+         components to use new, semantic names (specific names TBD). This will
+         not result in any visual changes. **This is likely to be the largest PR
+         to review as it is a global find/replace across the entire frontend.**
+
+         1. Replace the "hardcoded" color values in the Tailwind configuration
+            file with css variables defined in the "base" layer of the
+            `tailwind.css` file. This will not result in any visual changes.
+
+      2. Add the dark mode colors as CSS variable definitions, nested under the
+         `.dark-mode` CSS class, in the "base" layer of the `tailwind.css` file.
+         This will not result in any visual changes, _except when
+         `FORCE_DARK_MODE` is enabled._
+
+      3. Visual Regression tests. Update
+         `frontend/test/playwright/utils/breakpoints.ts` so that each breakpoint
+         produces and expects a dark mode screenshot to pass as well as the
+         existing light mode screenshot. **This will also be a significant diff,
+         as at the time of writing it will create 293 new screenshots to review.
+         This is also the point of the process where @fcoveram and
+         @wordpress/openverse-frontend should review the full dark mode
+         appearance for correctness and sufficient color contrast (see the
+         ["Accessibility"](#accessibility) section for more details.**
```

Comment:
Small detail question. Should the old snapshots be renamed to `<snapshot_name>_light` so that the new ones are similarly `_dark`? Or should the old ones retain their names, and only new ones  have an additional suffix? If the former, it will introduce a significant diff in old snapshots in addition to the dark mode ones, so it would be better to do in a separate PR for ease of review.

Reply:
I don't have a strong preference either way.

Reply:
Renaming existing snapshots should just have file-rename diffs, not diffs of the actual content (I hope). Adding the colour scheme to all snapshots explicitly sounds like a good idea to keep things easy to understand and maximally clear.

Reply:
Added in 6295bb0a496428a19e22329f4dd4e1c863a60f17